apm-sniffer/apm-agent/src/main/java/org/apache/skywalking/apm/agent/SkyWalkingAgent.java

```diff
@@ -203,4 +213,25 @@ public void onError(final String typeName,
         public void onComplete(String typeName, ClassLoader classLoader, JavaModule module, boolean loaded) {
         }
     }
+
+    static void configureLogger() {
```

Comment:
Please read the existing source codes, all config should be controlled through `Config`, which has supported dynamic injection already.

Reply:
The problem here is now the resolver is set to `PatternResolver` by default.

We have to setup the resolver before any `LogManager.getLogger(...)` is called. So this method is supposed to used in the static block of `SkywalkingAgent`.

Reply:
I am not saying this, take a closer look about how we initialize the Config's fields.

Reply:
> I am not saying this, take a closer look about how we initialize the Config's fields.

Yes. I understand the dynamic injection is implemented in `SnifferConfigInitializer` with the help of `ConfigInitializer`. 

Now I take advantage of `ConfigInitializer`, in which the `logger` is declared but actually not used, to inject the system properties, though limitation is still there.

Reply:
The variable should not be injected through the way you did.

Reply:
> The variable should not be injected through the way you did.

Well. I remove this static block and first init config as it previously did. Then I reconfigure the `Logger` after config initialization finishes.

The procedure now completely follows the rules except the first few log calls in `SnifferConfigInitializer` will still use the default logger.